pkg/reconciler/autoscaling/hpa/hpa_test.go

```diff
@@ -90,6 +93,7 @@ func TestControllerCanReconcile(t *testing.T) {
 }
 
 func TestReconcile(t *testing.T) {
+	retryAttempted := make(map[string]bool)
```

Comment:
It seems the table test cases are run sequentially. Would it work if we make this a bool and reset it back to `false` in the reactor function under the condition `retryAttempted && action.Matches("update", "podautoscalers")`?

Reply:
I concur with @taragu's comment, but the "fix" needs to look a little different. I played with this a little to not send you down a wrong path, here's a suggested diff:

```diff
diff --git a/pkg/reconciler/autoscaling/hpa/hpa_test.go b/pkg/reconciler/autoscaling/hpa/hpa_test.go
index df25b2099..1e5a5c740 100644
--- a/pkg/reconciler/autoscaling/hpa/hpa_test.go
+++ b/pkg/reconciler/autoscaling/hpa/hpa_test.go
@@ -93,7 +93,7 @@ func TestControllerCanReconcile(t *testing.T) {
 }
 
 func TestReconcile(t *testing.T) {
-	retryAttempted := make(map[string]bool)
+	retryAttempted := false
 	const (
 		deployName = testRevision + "-deployment"
 		privateSvc = testRevision + "-private"
@@ -135,10 +135,11 @@ func TestReconcile(t *testing.T) {
 		Key: key(testNamespace, testRevision),
 		WithReactors: []ktesting.ReactionFunc{
 			func(action ktesting.Action) (handled bool, ret runtime.Object, err error) {
-				if retryAttempted["create hpa & sks, with retry"] || !action.Matches("update", "podautoscalers") {
+				if !action.Matches("update", "podautoscalers") || action.GetSubresource() != "status" || retryAttempted {
 					return false, nil, nil
 				}
-				retryAttempted["create hpa & sks, with retry"] = true
+
+				retryAttempted = true
 				return true, nil, apierrs.NewConflict(v1alpha1.Resource("foo"), "bar", errors.New("foo"))
 			},
 		},
@@ -457,6 +458,7 @@ func TestReconcile(t *testing.T) {
 	}}
 
 	table.Test(t, MakeFactory(func(ctx context.Context, listers *Listers, cmw configmap.Watcher) controller.Reconciler {
+		retryAttempted = false
 		ctx = podscalable.WithDuck(ctx)
 
 		return &Reconciler{
```

1. Yes we can make this a boolean to make the actual tests a little less verbose. We can reset it before each testrun though. Otherwise we might end up in a spinning loop.
2. I'd narrow down the action check to also include the subresource. It's a bummer the `Matches` call down't allow for the `podautoscalers/status` syntax. Gonna file a PR upstream to add that.

Reply:
As promised 😂 https://github.com/kubernetes/kubernetes/pull/87687

Reply:
thanks for the detailed feedback. I made the var a boolean and made sure its reset in each tests and added the check for the subresource (as your PR is still pending, but let me know if you want me to wait till your PR is in and then update here).